src/covid_shared/data_transformations.py

```diff
@@ -0,0 +1,58 @@
+import pandas as pd
+
+
+def make_round_trip_invariant(cumulative_time_series: pd.Series) -> pd.Series:
+    """Drops nulls, interpolates, and adds a leading zero.
+
+    This function drops nulls from the cumulative time series, interpolates
+    missing days linearly, and then pads the front of the
+    series with a 0 so that the resulting series has the property:
+
+        s == s.groupby('location_id').diff().fillna(0).groupby('location_id').cumsum()
+
+    which is to say we can preserve the counts under conversions between daily
+    and cumulative space.
```

Comment:
It is good you wrote these 2 lines, because I did not derive that from the 5 statements in that line of code.